### ISSUE TITLE
PHP 7.4 - false positive unnecessary null

### ISSUE THREAD

**Issue by janlanger** [kalessil/phpinspectionsea#1457]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 4.0.1.1  |
| Language level | PHP 7.4                        |

Current behaviour
--
```
class Foo {
    private ?DateTimeImmutable $bar = null;
    
    public function get(): DateTimeImmutable
    {
        if ($this->bar !== null) { $this->bar = new DateTimeImmutable(); }
        
        return $this->bar;
    }
}
```

In order for this code to work, $bar variable must be explicitly initialized with null in PHP 7.4. But currently, inspections marks this `= null` as unnecessary.

Expected behavior
--
Initializing property with nullable typehint with null should not be marked unnecessary. For non-static private properties it could be marked unnecessary only if the value is always set in constructor.

**Comment by orklah** [comment]
+1

For reference, there was an attempt to fix the issue already:
https://github.com/kalessil/phpinspectionsea/issues/1380#issuecomment-519503515

**Comment by kalessil** [comment]
Hmm, this supposes to be fixed. I'll check!

**Comment by kalessil** [comment]
Thank you for reporting @janlanger. This ticket duplicates #1455.

**Comment by kalessil** [comment]
Fixed in #1455 !
